Hestia/Hestia/DevicesScreen/AddDeviceScreen/TableSourceAddDeviceManufacturer.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using UIKit;
+using Foundation;
+
+using Hestia.DevicesScreen.resources;
+using System.Drawing;
+using System.Collections;
+using Hestia.backend;
+using Hestia.backend.models;
+
+namespace Hestia.DevicesScreen
+{
+    public class TableSourceAddDeviceManufacturer : UITableViewSource
+    {
+        // The viewController to which the TableView connected to this Source lives in
+        UITableViewControllerAddDevice owner;
+
+        // Hashtable of plugins (keyed on collection)
+        Hashtable plugins;
+
+        // The list with collections, set in the constructor. (Retrieved from server)
+        List<string> collections;
+
+        // The kind of cell that is used in the table (set in Storyboard)
+        string CellIdentifier = "tableCellManufacturer";
+
+        // Constructor. Gets the device data and the ViewController
+        public TableSourceAddDeviceManufacturer(List<string> collections,
+                                    Hashtable plugins,
+                    UITableViewControllerAddDevice owner)
+        {
+            this.plugins = plugins;
+            this.owner = owner;
+            this.collections = collections;
+        }
+
+        // We have only one section with manufacturers
+        public override nint NumberOfSections(UITableView tableView)
+        {
+            return 1;
```

Comment:
Same, if the number of sections changes you have to change all files, and it isn't neat to do this.